appGeo/src/app/game/game-rules.component.ts

```diff
@@ -0,0 +1,202 @@
+import { Component } from '@angular/core';
+
+export const SUCCESS = 10
+export const FAILURE = -10
+
+const MIN_MIN_PLAYERS = 5
+const MIN_FRACTION_KILLERS = 0.2
+
+//all constants in minutes
+const MIN_GAME_LENGTH = 1440
+const DEF_GAME_LENGTH = 10080
+
+//safe length default is 1/3 of game length. Minimum is 1/2 of game length
+
+const DEF_CYCLE_LEN = 1440
+
+//cannot be longer than safe length
+const MIN_VOTE_LENGTH = 60
+
+//minutes after game begins
+const DEF_VOTE_TIME = 960
+
+const MIN_MAX_KILL = 1
+const DEF_MAX_GLOBAL_KILL = Infinity
+
+@Component({
+    selector: 'game-rules',
+    templateUrl: './game-rules.component.html',
+    styleUrls: ['./game-rules.component.css']
+})
+export class GameRules {
+
+    testing_overrule: boolean //default false. Turn on if lengths need to be shortened
+
+    scheduledEnd: boolean //if the game has a scheduled end date
+    wipeOutEnd: boolean //if true then killers need to kill all civilians to win, or just get majority
+
+    minPlayers: number //number of minimum players
+    fractionKillers: number //fractional ratio of civilians who are killers
+
+    gameLength: number //the number of minutes the game is long.
+    dayCycleLength: number //the number of minutes in the day cycle
+    safeLength: number //the number of minutes kills aren't allowed each cycle
+    voteLength: number //the number of minutes at the begining of safe time voting is allowed
+    
+    //number of minutes after midnight
+    voteTime: number //number of minutes 
+
+    maxSoloKills: number //number of kills a single killer can do a day
+    maxGlobalKills: number //number of kills per day allowed total
+
+    constructor(gameLengthHours?) {
```

Comment:
This is causing an issue with compiling, what is `gameLengthHours?`

Reply:
It's meant to be an optional variable. I'm cutting it now to make compiling easier, which will fix that